src/itx.rs

```diff
@@ -0,0 +1,141 @@
+use crate::include::common::bitdepth::AsPrimitive;
+use crate::include::common::bitdepth::BitDepth;
+use crate::include::common::bitdepth::FromPrimitive;
+use crate::include::common::bitdepth::ToPrimitive;
+use crate::include::common::intops::iclip;
+use crate::include::common::intops::imin;
+use crate::include::stddef::*;
+use crate::include::stdint::*;
+
+extern "C" {
+    fn memset(_: *mut libc::c_void, _: libc::c_int, _: libc::c_ulong) -> *mut libc::c_void;
+}
+pub type itx_1d_fn =
+    Option<unsafe extern "C" fn(*mut int32_t, ptrdiff_t, libc::c_int, libc::c_int) -> ()>;
+
+pub unsafe extern "C" fn inv_txfm_add_rust<BD: BitDepth>(
+    mut dst: *mut BD::Pixel,
+    stride: ptrdiff_t,
+    coeff: *mut BD::Coef,
+    eob: libc::c_int,
+    w: libc::c_int,
+    h: libc::c_int,
+    shift: libc::c_int,
+    first_1d_fn: itx_1d_fn,
+    second_1d_fn: itx_1d_fn,
+    has_dconly: libc::c_int,
+    bd: BD,
+) {
+    let bitdepth_max: libc::c_int = bd.bitdepth_max().to_prim();
```

Comment:
We shouldn't need to use the `.{to,from}_prim()` methods.  I think the `.as_()` method suffices (let me know if it doesn't work) and is the one meant to be used, whereas the prim ones are more for the implementation.